modules/vue-i18n/composables/useLocalePath.ts

```diff
@@ -13,27 +16,31 @@ export default () => {
     const selectedLocale = VUE_I18N_LOCALES.find((l) => l.name === locale)!
 
     const fullPath = (): string => {
-      const currentPrefix = selectedLocale.prefix
-
-      if (typeof target === 'string') {
-        if (target.startsWith('/')) {
-          if (currentPrefix === '/') {
-            return router.resolve(target).fullPath
-          } else {
-            return router.resolve(`${currentPrefix}${target}`).fullPath
-          }
-        } else {
-          return router.resolve({ name: `${target}__${locale}` }).fullPath
+      let originalRouteTarget = {}
+      let sanitizedPath
+      const currentLocalePrefix = selectedLocale.prefix === '/' ? '' : selectedLocale.prefix
+
+      if (isString(target)) {
+        sanitizedPath = target.startsWith('/') ? target : `/${target}`
+        if (target === 'index') {
+          sanitizedPath = '/'
         }
-      } else if (isNamed(target)) {
-        const [name] = target.name!.toString().split('__')
-
-        return router.resolve({ ...target, name: `${name}__${locale}` }).fullPath
+      } else if (isObject(target) && 'name' in target) {
+        const { name, ...sanitazedTarget } = target
+        sanitizedPath = `/${String(name)}`
```

Comment:
I'm afraid this isnt true for routes with parameters. E.g. `/product/[id].vue`. Name if this route is `product/[id]` but thats definetelly not the path.

Reply:
We should ale add tests for these parametrized routes.